src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -3,13 +3,17 @@
 import java.time.LocalDate;
 import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import wolox.training.models.Book;
 import wolox.training.models.User;
 
 public interface UserRepository extends JpaRepository<User, Long> {
 
     Optional<User> findFirstByUsernameOrderByIdAsc(String username);
 
-    Iterable<User> findAllByBirthdateBetweenAndNameIgnoreCaseContaining(LocalDate birthdateIni,
-            LocalDate birthdateEnd, String name);
-
+    @Query("SELECT u FROM User u WHERE (:bi is null or u.birthdate >= :bi) and "
+        + "(:bf is null or u.birthdate <= :bf) and (:na <= '' or upper(u.name) = upper(:na))")
```

Comment:
La comparación de fechas se podría facilitar con la sentencia between, y el nombre también se compararía contra null y no contra empty. Se podría hacer algo como esto: 
```suggestion
	@Query("select u from User u where ((cast(:bi as date) is null or cast(:bf as date) is null or"
	        + " u.birthdate between :bi and :bf) and (:na is null or lower(u.name) like lower(concat('%',:na,'%'))))")
```

Reply:
Buenisimo!!! ahi lo modifique!